lib/design_codes/utils/code_requirement.rb

```diff
@@ -17,6 +17,9 @@ module ClassMethods
         def call(params = {})
           schema_klass.validate!(params)
           sanitized_params = schema_klass.structurize(params)
+          sanitized_params.members.each do |param_name|
+            define_method(param_name) { sanitized_params[param_name] }
+          end
```

Comment:
this addition to the `CodeRequirement` module will programmatically define getter methods for each sanitized param that is being passed. That way if we pass a param called `design_compression_strength`, we can get its value by either using `params.design_compression_strength`, or `design_compression_strength` directly